### PR TITLE
Move `valid_categories` to `particle_class.py`

### LINKED ISSUE #1719: Change `Particle.is_category.valid_categories` to `plasmapy.particles.particle_class.valid_categories`
Following a discussion on #1710.

### PR DESCRIPTION
**Pull request by namurphy**
This PR follows a suggestion from @rocco8773 to move `AbstractPhysicalParticle.is_category.valid_categories` elsewhere.  I decided to put it in `plasmapy.particles.particle_class.valid_categories` as its own thing.

We could make it a `frozenset` pretty straightforwardly, but if someone wanted to add their own particle categories, they would not be able to do so, so I kept it as a `set`.

Closes #1719.

### CODE REVIEW

**Comment by namurphy** [comment]
pre-commit.ci autofix

File: plasmapy/particles/particle_class.py [hunk 1]
```diff
@@ -74,12 +75,24 @@
...
     | _atomic_property_categories
     | _specific_particle_categories
 )
+r"""
+A `set` containing all valid particle categories.
+
+To add a new particle category, update
+`~plasmapy.particles.particle_class.valid_categories` with a new string
+representing the category (e.g., by doing
+:py:`valid_categories |= {"new_category"}`\ ).
```
Comment (namurphy) [inline plasmapy/particles/particle_class.py:90]:
```suggestion
```

File: docs/notebooks/getting_started/particles.ipynb [hunk 2]
```diff
@@ -748,10 +749,9 @@
    "id": "a03c8153",
    "metadata": {},
    "source": [
-    "[is_category()]: ../../api/plasmapy.particles.particle_class.Particle.rst#plasmapy.particles.particle_class.Particle.is_category\n",
-    "[Particle]: ../../api/plasmapy.particles.particle_class.Particle.rst\n",
+    "[`valid_categories`]: ../../api/plasmapy.particles.particle_class.valid_categories.rst\n",
```
Comment (namurphy) [inline docs/notebooks/getting_started/particles.ipynb:752]:
```suggestion
    "[valid_categories]: ../../api/plasmapy.particles.particle_class.valid_categories.rst\n",
```

File: docs/notebooks/getting_started/particles.ipynb [hunk 3]
```diff
@@ -748,10 +749,9 @@
...
     "\n",
-    "The `valid_categories` attribute of [is_category()] for any [Particle] gives a set containing all valid categories."
+    "All valid particle categories are included in [`valid_categories`]."
```
Comment (namurphy) [inline docs/notebooks/getting_started/particles.ipynb:754]:
```suggestion
    "All valid particle categories are included in [valid_categories]."
```

**Review by rocco8773 — APPROVED** [review]

**Comment by namurphy** [comment]
Before merging, need to make sure that the `valid_categories` link in the example notebook shows up okay.

**Pull request merged**